File: 01/two.py

def calc(input, total=0, freq={}):
    found = None

    for item in input:
        if total in freq:
            print(total)
            found = total
            break
        else:
            freq[total] = True

        sign = item[0]
        num = int(item[1:])
        if sign == '+':
            total += num
        else:
            total -= num

    if not found:
        return calc(input, total)

    return found


if __name__ == "__main__":
    with open("input.txt") as file:
        data = file.read().split('\n')
        print(calc(data))

File: 01/two.js
const fs = require('fs');

function calc(input, total = 0, freq = {}) {
  let found;

  for (const item of input) {
    if (total in freq) {
      found = total;
      break;
    } else {
      freq[total] = true;
    }
    total += +item;
  }

  if (!found) {
    return calc(input, total, freq);
  }

  return found;
}

const input = fs
  .readFileSync('input.txt')
  .toString()
  .split('\n');
const output = calc(input);
console.log('FOUND:', output);
